middleware/validationMiddleware.js

```diff
@@ -0,0 +1,18 @@
+const Ajv = require("ajv");
+const ajvFormats = require("ajv-formats");
+
+const ajv = new Ajv();
+ajvFormats(ajv);
+
+const validateSchema = (schema) => {
+  const validate = ajv.compile(schema);
+  return (req, res, next) => {
+    const valid = validate(req.body);
```

Comment:
this is limited to only request body validation, rather it should be generic for working with request queries as well